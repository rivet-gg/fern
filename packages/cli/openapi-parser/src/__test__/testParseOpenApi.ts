import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
