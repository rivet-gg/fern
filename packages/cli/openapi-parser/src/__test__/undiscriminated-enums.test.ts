import { testParseOpenAPI } from "./testParseOpenApi";

describe("open api parser", () => {
    testParseOpenAPI("undiscriminated-enums", "openapi.yml");
});
