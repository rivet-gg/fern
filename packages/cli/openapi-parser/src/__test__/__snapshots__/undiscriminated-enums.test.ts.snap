// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api parser undiscriminated-enums parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [],
      "generatedRequestName": "GetExampleRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": null,
      "pagination": null,
      "path": "/example",
      "pathParameters": [],
      "queryParameters": [],
      "request": null,
      "requestNameOverride": null,
      "response": {
        "description": "Successful response",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "GetExampleResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "ExampleResponse",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": "Get Example",
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [
    "ExampleResponse",
    "GreetingEnum",
    "FarewellEnum",
  ],
  "schemas": {
    "Conversation": {
      "type": "oneOf",
      "value": {
        "availability": null,
        "commonProperties": [],
        "description": null,
        "discriminantProperty": "type",
        "generatedName": "Conversation",
        "groupName": null,
        "nameOverride": null,
        "schemas": {
          "goodbye": {
            "availability": null,
            "description": null,
            "generatedName": "ComponentsSchemasFarewell",
            "groupName": null,
            "nameOverride": null,
            "schema": "Farewell",
            "type": "reference",
          },
          "hello": {
            "availability": null,
            "description": null,
            "generatedName": "ComponentsSchemasGreeting",
            "groupName": null,
            "nameOverride": null,
            "schema": "Greeting",
            "type": "reference",
          },
        },
        "type": "discriminated",
      },
    },
    "Farewell": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "Farewell",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "farewellMessage",
          "key": "message",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "farewellMessage",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "FarewellMessage",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "FarewellEnum": {
      "availability": null,
      "default": null,
      "description": null,
      "generatedName": "FarewellEnum",
      "groupName": null,
      "nameOverride": null,
      "type": "enum",
      "values": [
        {
          "availability": null,
          "casing": {
            "camel": null,
            "pascal": null,
            "screamingSnake": null,
            "snake": null,
          },
          "description": null,
          "generatedName": "goodbye",
          "nameOverride": null,
          "value": "goodbye",
        },
        {
          "availability": null,
          "casing": {
            "camel": null,
            "pascal": null,
            "screamingSnake": null,
            "snake": null,
          },
          "description": null,
          "generatedName": "GOODBYE",
          "nameOverride": null,
          "value": "GOODBYE",
        },
      ],
    },
    "Greeting": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "Greeting",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "greetingMessage",
          "key": "message",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "greetingMessage",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GreetingMessage",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "GreetingEnum": {
      "availability": null,
      "default": null,
      "description": null,
      "generatedName": "GreetingEnum",
      "groupName": null,
      "nameOverride": null,
      "type": "enum",
      "values": [
        {
          "availability": null,
          "casing": {
            "camel": null,
            "pascal": null,
            "screamingSnake": null,
            "snake": null,
          },
          "description": null,
          "generatedName": "hello",
          "nameOverride": null,
          "value": "hello",
        },
        {
          "availability": null,
          "casing": {
            "camel": null,
            "pascal": null,
            "screamingSnake": null,
            "snake": null,
          },
          "description": null,
          "generatedName": "HELLO",
          "nameOverride": null,
          "value": "HELLO",
        },
      ],
    },
  },
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Example API",
  "variables": {},
  "webhooks": [],
}
`;
